protocol-impl/src/main/java/io/camunda/zeebe/protocol/impl/stream/job/JobActivationPropertiesImpl.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.protocol.impl.stream.job;
+
+import io.camunda.zeebe.msgpack.UnpackedObject;
+import io.camunda.zeebe.msgpack.property.ArrayProperty;
+import io.camunda.zeebe.msgpack.property.LongProperty;
+import io.camunda.zeebe.msgpack.property.StringProperty;
+import io.camunda.zeebe.msgpack.value.StringValue;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.agrona.DirectBuffer;
+
+public class JobActivationPropertiesImpl extends UnpackedObject implements JobActivationProperties {
+  private final StringProperty workerProp = new StringProperty("worker", "");
+  private final LongProperty timeoutProp = new LongProperty("timeout", -1);
+  private final ArrayProperty<StringValue> fetchVariablesProp =
+      new ArrayProperty<>("variables", new StringValue());
+
+  public void JobActivationProperties() {
+    declareProperty(workerProp).declareProperty(timeoutProp).declareProperty(fetchVariablesProp);
+  }
+
+  public JobActivationPropertiesImpl setWorker(
+      final DirectBuffer worker, final int offset, final int length) {
+    workerProp.setValue(worker, offset, length);
+    return this;
+  }
+
+  public JobActivationPropertiesImpl setTimeout(final long val) {
+    timeoutProp.setValue(val);
+    return this;
+  }
+
+  public JobActivationPropertiesImpl setFetchVariables(final List<StringValue> variables) {
+    fetchVariablesProp.reset();
+    variables.forEach(variable -> fetchVariablesProp.add().wrap(variable));
+    return this;
+  }
+
+  @Override
+  public DirectBuffer worker() {
+    return workerProp.getValue();
+  }
+
+  @Override
+  public Collection<DirectBuffer> fetchVariables() {
+    return fetchVariablesProp.stream().map(StringValue::getValue).collect(Collectors.toList());
```

Comment:
:x: The way ArrayValue iteration works reuses the same buffer, so what's happening here is that you end up with a list with the same buffer three times.

You can fix it by wrapping the `StringValue::getValue` into its own `UnsafeBuffer` like so:

```suggestion
    return fetchVariablesProp.stream()
        .map(val -> new UnsafeBuffer(val.getValue()))
        .collect(Collectors.toList());
```

Reply:
That's a pretty obscure "feature" (or bug :smile:) in our message pack implementation. This isn't the first time someone runs into this (I know only because it happened to me too :smile:), so do raise it to your team about changing this behavior. It's maybe a very small optimization for too much headache.

Reply:
Good catch @npepinpe, that helped me tho 😅